lib/util/source-code.js

```diff
@@ -88,68 +88,66 @@ function looksLikeExport(astNode) {
  * @param {ASTNode} ast - The Program node of the AST representing the code. This AST should be created from the text that BOM was stripped.
  * @constructor
  */
-function SourceCode(text, ast) {
-    validate(ast);
+class SourceCode {
```

Comment:
Thanks for the link, I learned some new things about classes. :+1:

As far as externally-observable behavior is concerned, it looks like the relevant differences are: 

* All methods are non-enumerable
* All methods throw with `new`
* Calling the class constructor without `new` throws an error.

(5) is not actually applicable to this case because we assign to `this` in strict mode, so `SourceCode` would throw an error when called without `new` anyway. But I agree that the other two are breaking if someone was doing something weird such as `new (context.getSourceCode().getText)()`.